plex/api.py

```diff
@@ -665,8 +676,21 @@ def fillVideoInfos(
 
         if mediaPart:
             # extract the absolute / actual path and the stream URL from the selected MediaPart
-            info['path'] = mediaPart.file
-            item.setPath(plexServer.url(mediaPart.key, includeToken=True))
+            info['path'] = mediaPart.file  # path to file on disk
+
+            # determine if directPlay is enabled and possible
+            if allowDirectPlay:
+                # perform replacement in file path if configured
+                filePath = Api._mapPath(mediaPart.file)
+
+                # confirm file is accessible, fall back to streaming otherwise
+                if filePath and xbmcvfs.exists(filePath):
+                    item.setPath(filePath)
+                else:
+                    log(f"Direct play configured but path {filePath} is inaccessible.", xbmc.LOGWARNING)
+                    item.setPath(plexServer.url(mediaPart.key, includeToken=True))
```

Comment:
This line appears twice now. Maybe don't do anything here and in the `else` below and then check if `item.getPath()` is empty and fall back to using `item.setPath(plexServer.url(mediaPart.key, includeToken=True))` then to avoid code duplication.